libutils/mustache.c

```diff
@@ -447,14 +457,54 @@ static bool RenderVariable(Buffer *out,
     switch (JsonGetElementType(var))
     {
     case JSON_ELEMENT_TYPE_PRIMITIVE:
-        // note that this also covers 'serialize' on primitives
-        return RenderVariablePrimitive(out, var, escape, key_mode);
+        if (key_mode && SeqLength(hash_stack) > 1)
+        {
+            JsonElement *parent = SeqAt(hash_stack, SeqLength(hash_stack) - 2);
+
+            if (JsonGetElementType(parent) == JSON_ELEMENT_TYPE_CONTAINER &&
+                JsonGetContainerType(parent) == JSON_CONTAINER_TYPE_ARRAY)
+            {
+                BufferAppendF(out, "%jd", iteration_position);
+                return true;
+            }
+            else
+            {
+                return RenderVariablePrimitive(out, var, escape, key_mode);
+            }
+        }
+        else
+        {
+            // note that this also covers 'serialize' on primitives
+            return RenderVariablePrimitive(out, var, escape, key_mode);
+        }
```

Comment:
You can get rid of both else blocks here and consolidate the two ``RenderVariablePrimitive`` into one.